File: javascript/leetcode/insert-delete-getrandom-o1-duplicates-allowed.js
/**
 * Initialize your data structure here.
 */
var RandomizedCollection = function() {
    this.length = 0
    this.set = []
};

/**
 * Inserts a value to the set. Returns true if the set did not already contain the specified element. 
 * @param {number} val
 * @return {boolean}
 */
RandomizedCollection.prototype.insert = function(val) {
    doesNotExist = true
    for (var i = 0; i < this.length; i++) {
        if (this.set[i] == val) {
            doesNotExist = false
            break
        }
    }
    this.set.push(val)
    this.length++
    return doesNotExist
};

/**
 * Removes a value from the set. Returns true if the set contained the specified element. 
 * @param {number} val
 * @return {boolean}
 */
RandomizedCollection.prototype.remove = function(val) {
    for (var i = 0; i < this.length; i++) {
        if (this.set[i] == val) {
            this.set.splice(i, 1)
            this.length--
            return true
        }
    }
    return false
};

/**
 * Get a random element from the set.
 * @return {number}
 */
RandomizedCollection.prototype.getRandom = function() {
    return this.set[Math.floor(Math.random() * this.length)]
};

/** 
 * Your RandomizedSet object will be instantiated and called as such:
 * var obj = Object.create(RandomizedSet).createNew()
 * var param_1 = obj.insert(val)
 * var param_2 = obj.remove(val)
 * var param_3 = obj.getRandom()
 */
 

File: python/leetcode/insert-delete-getrandom-o1-duplicates-allowed.py
class RandomizedCollection:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.set = []
        self.length = 0

    def insert(self, val):
        """
        Inserts a value to the set. Returns true if the set did not already contain the specified element.
        :type val: int
        :rtype: bool
        """
        does_not_exist = True
        for n in self.set:
            if n == val:
                does_not_exist = False
                break
        self.set.append(val)
        self.length += 1
        return does_not_exist

    def remove(self, val):
        """
        Removes a value from the set. Returns true if the set contained the specified element.
        :type val: int
        :rtype: bool
        """
        for i in range(self.length):
            if self.set[i] == val:
                self.length -= 1
                del self.set[i]
                return True
        return False
        

    def getRandom(self):
        """
        Get a random element from the set.
        :rtype: int
        """
        return self.set[int(random.random() * self.length // 1)]
        


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()
